### PR TITLE
Web graphics slw

### PR DESCRIPTION
**Pull request by cshartsough**
These changes are to create a supercooled liquid water (SLW) plot.

The main item remaining for this is to switch to `wrfnat` from `wrfprs` grib2 data.  A few smaller issues, such as the need to give a vertical level in the data calls even with the one_lev option set to False, and a pylint "unused parameter" warning (currently disabled).

Comparison plots:

slw_sfc_f18

Screen Shot 2021-01-28 at 1 06 00 PM

### CODE REVIEW

File: adb_graphics/default_specs.yml [hunk 1]
```diff
@@ -992,6 +1010,7 @@ temp: # Temperature
       pres_sfc:
         hatches: ['', '...']
         levels: [0, 925]
+#    ncl_name: TMP_P0_L105_GLC0 # 2m Temp - wrfnat test
```
Comment (christinaholtNOAA) [inline adb_graphics/default_specs.yml:1013]: This one shouldn't be needed since it's included already for both wrfprs and wrfnat under the ua_temp anchor.

**Review by christinaholtNOAA — COMMENTED** [review]
This looks good! I can't see any obvious reasons this wouldn't work for a wrfnat file. Let's chat if you are still running into issues when trying out nat files.

**Review by Brian-Jamison — APPROVED** [review]
This looks good!  I hope using the nat file won't be too painful.

File: adb_graphics/datahandler/grib.py [hunk 2]
```diff
@@ -417,6 +417,25 @@ def grid_info(self):
 
         return grid_info
 
+    def supercooled_liquid_water(self, values) -> np.ndarray:
```
[the remark this answers is not in the document]
Comment (cshartsough) [reply adb_graphics/datahandler/grib.py]: added **kwargs

File: adb_graphics/datahandler/grib.py [hunk 3]
```diff
@@ -441,11 +460,13 @@ def values(self, level=None, name=None, **kwargs) -> np.ndarray:
 
         Optional Input:
             name       the name of a field other than defined in self
+            one_lev    bool flag. if True, get the single level of the variable
```
[the remark this answers is not in the document]
Comment (cshartsough) [reply adb_graphics/datahandler/grib.py:512]: was able to put it in asI resolved conflicts.

File: adb_graphics/default_specs.yml [hunk 4]
```diff
@@ -297,6 +297,10 @@ cloudcover:
     <<: *cld_cover
     ncl_name: TCDC_P0_L10_{grid}
     title: Total Cloud Cover
+clwmr: # Cloud water Mixing Ratio
+  850mb:
+    ncl_name: CLWMR_P0_L100_{grid}
+#    ncl_name: CLWMR_P0_L105_GLC0  # wrfnat test
```
[the remark this answers is not in the document]
Comment (cshartsough) [reply adb_graphics/default_specs.yml]: added the {grid} option

File: adb_graphics/default_specs.yml (hunk 1 above)
Comment (cshartsough) [reply adb_graphics/default_specs.yml]: this was fixed when I added the wrfnat option in create_graphics.

File: adb_graphics/datahandler/grib.py [hunk 5]
```diff
@@ -463,11 +484,13 @@ def values(self, level=None, name=None, **kwargs) -> np.ndarray:
...
-            if vertical_index is None:
-                lev = self.get_level(field, level, spec)
-            vals = field[lev, :, :]
+            if one_lev:
+                lev = vertical_index
+                if vertical_index is None:
+                    lev = self.get_level(field, level, spec)
+                vals = field[lev, :, :]
+            else:
+                vals = field[:, :, :]
```
[the remark this answers is not in the document]
Comment (cshartsough) [reply adb_graphics/datahandler/grib.py:549]: :-)

File: adb_graphics/datahandler/grib.py [hunk 6]
```diff
@@ -451,6 +451,36 @@ def grid_info(self):
...
+        ps = self.values(name='pres', level='sfc') * 100. # convert back to Pa
+        p = self.values(name='pres', level='ua', one_lev=False)
+        t = self.values(name='temp', level='ua', one_lev=False)
+        cmr = self.values(name='clwmr', level='ua', one_lev=False)
+        rmr = self.values(name='rwmr', level='ua', one_lev=False)
+
+        g = 9.81 # gravity
+        slw = ps * 0.
+
+        nlevs = np.shape(p)[0]
```
Comment (christinaholtNOAA) [inline adb_graphics/datahandler/grib.py:471]: Maybe here a comment on the algorithm. Maybe that's something like:

>  Compute pressure on native levels by iteratively adding the average change in pressure over the layer to the known pressure at the next lower level, starting at the surface.

File: adb_graphics/datahandler/grib.py [hunk 7]
```diff
@@ -451,6 +451,36 @@ def grid_info(self):
...
+        for n in range(nlevs):
+            if n == 0:
+                dp = 2 * (ps[:, :] - p[n, :, :])
+                Pm = ps - dp
+            else:
+                dp = 2 * (Pm[:, :] - p[n, :, :])
+                Pm = Pm - dp
```
Comment (christinaholtNOAA) [inline adb_graphics/datahandler/grib.py:478]: A comment on what the next two lines are dong would be helpful.

I'm guessing "Set upper boundary of slw to 0 when temperature is above freezing."

I'm not even sure how to guess on what the next line is doing. Scaling it based on the pressure change?

File: adb_graphics/datahandler/grib.py [hunk 8]
```diff
@@ -451,6 +451,36 @@ def grid_info(self):
...
+        Generates a field of Supercooled Liquid Water
+        '''
+
+        ps = self.values(name='pres', level='sfc') * 100. # convert back to Pa
+        p = self.values(name='pres', level='ua', one_lev=False)
+        t = self.values(name='temp', level='ua', one_lev=False)
+        cmr = self.values(name='clwmr', level='ua', one_lev=False)
+        rmr = self.values(name='rwmr', level='ua', one_lev=False)
+
+        g = 9.81 # gravity
```
Comment (christinaholtNOAA) [inline adb_graphics/datahandler/grib.py:468]: Why this value for gravity? Should we be using a value that varies with height?

**Review by christinaholtNOAA — COMMENTED** [review]
The code implementation looks good! Just a few comments on some additional documentation that could help our future selves, and maybe a couple of tiny fixes that were introduced during testing, I'm sure.

File: adb_graphics/default_specs.yml [hunk 9]
```diff
@@ -296,6 +296,16 @@ cloudcover:
     <<: *cld_cover
     ncl_name: TCDC_P0_L{level_type}_{grid}
     title: Total Cloud Cover
+clwmr: # Cloud water Mixing Ratio
+  ua:
+    ncl_name: 
+      nat: CLWMR_P0_L105_{grid}
+      prs: CLWMR_P0_L100_{grid}
+cref: # Composite reflectivity
```
[the remark this answers is not in the document]
Comment (cshartsough) [reply adb_graphics/default_specs.yml:304]: no, operator error resolving conflict. I removed the extra cref listing.

File: adb_graphics/datahandler/grib.py [hunk 10]
```diff
@@ -451,6 +451,36 @@ def grid_info(self):
...
         return grid_info
 
+    def supercooled_liquid_water(self, values, **kwargs) -> np.ndarray:
+
+        # pylint: disable=unused-argument
+
+        '''
+        Generates a field of Supercooled Liquid Water
+        '''
+
```
[the remark this answers is not in the document]
Comment (cshartsough) [reply adb_graphics/datahandler/grib.py:461]: I added comments and more readable variables.

File: adb_graphics/datahandler/grib.py (hunk 6 above)
Comment (cshartsough) [reply adb_graphics/datahandler/grib.py:471]: added comments.

File: adb_graphics/datahandler/grib.py (hunk 7 above)
Comment (cshartsough) [reply adb_graphics/datahandler/grib.py:478]: added comments

File: adb_graphics/default_specs.yml [hunk 11]
```diff
@@ -630,7 +640,8 @@ pres:
     transform: conversions.pa_to_hpa
     unit: hPa
   ua:
-    ncl_name: PRES_P0_L105_{grid}
+    ncl_name:
+      nat: PRES_P0_L105_{grid}
```
[the remark this answers is not in the document]
Comment (cshartsough) [reply adb_graphics/default_specs.yml:644]: true, wrfprs only has mslp.

File: image_lists/hrrr_subset.yml [hunk 12]
```diff
@@ -1,174 +1,111 @@
 hourly:
```
[the remark this answers is not in the document]
Comment (cshartsough) [reply image_lists/hrrr_subset.yml:1]: another error from resolving conflicts.  replaced with old version.

File: adb_graphics/datahandler/grib.py (hunk 8 above)
Comment (cshartsough) [reply adb_graphics/datahandler/grib.py:468]: according to Curtis, a constant g of 9.81 m/s is fine.

**Review by christinaholtNOAA — APPROVED** [review]
Thanks for the quick turn around! Looks great!

**Pull request merged**